spec/install/deploy_spec.rb

```diff
@@ -295,6 +295,106 @@
       expect(out).not_to include("You have deleted from the Gemfile")
     end
 
+    it "allows to replace a host with the same host with credentials when allow_deployment_source_credential_changes is true" do
+      install_gemfile(<<-G, :artifice => "endpoint_strict_basic_authentication", :quiet => true)
+        source "http://user_name:password@localgemserver.test/"
+        gem "rack"
+      G
+
+      lockfile <<-G
+        GEM
+          remote: http://localgemserver.test/
+          specs:
+            rack (1.0.0)
+
+        PLATFORMS
+          #{local}
+
+        DEPENDENCIES
+          rack
+      G
+
+      bundle! "config allow_deployment_source_credential_changes true"
+      bundle :install, forgotten_command_line_options(:deployment => true)
+
+      expect(out).to match(/Could not reach host localgemserver.test/)
+    end
+
+    it "allows to replace a host with the same host with credentials when BUNDLE_ALLOW_DEPLOYMENT_SOURCE_CREDENTIAL_CHANGES env variable" do
```

Comment:
`when BUNDLE_ALLOW_DEPLOYMENT_SOURCE_CREDENTIAL_CHANGES env variable` this should be in a context block